src/Controller/PageAdminController.php

```diff
@@ -87,11 +87,11 @@ public function treeAction(Request $request): Response
         $pageManager = $this->container->get('sonata.page.manager.page');
 
         $currentSite = null;
-        $siteId = (int) $request->get('site');
+        $siteId = $request->get('site');
         foreach ($sites as $site) {
-            if ($siteId && $site->getId() === $siteId) {
+            if (null !== $siteId && $site->getId() === (int) $siteId) {
```

Comment:
```suggestion
            if (null !== $siteId && (string) $site->getId() === $siteId) {
```
Seems better because I think getId() can be string or int.